### ISSUE TITLE
matUtils extract results in Segmentation Fault

### ISSUE THREAD

**Issue by kvargha** [yatisht/usher#245]
Here are the files that are leading to a segmentation fault

- [Tree](https://storage.googleapis.com/ucsc-gi-cdph-bigtree/seqfault_tree/new_tree.pb)
- [VCF](https://storage.googleapis.com/ucsc-gi-cdph-bigtree/seqfault_tree/sample.vcf)
- [Samples file](https://storage.googleapis.com/ucsc-gi-cdph-bigtree/seqfault_tree/samples.txt)

Command that causes error:
`matUtils extract -i new_tree.pb  -s samples.txt -Y 5 -j output.json`

Causation Explanation From @AngieHinrichs :
It's line 1532 of mutation_annotated_tree.cpp (in get_subtree):
`                        if (curr->clade_annotations[k] != "") {`
Every node's clade_annotations array is expected to have the same size as the root node's clade_annotations array, with all items initialized to "" if they don't have an actual annotation on that node.  But sometimes some of them end up missing instead of "".

**Comment by jmcbroome** [comment]
The simple solution is just to iterate to the size of the annotations vector of the node being copied, instead of the number of annotations on the root. Since the new node having its information copied from the current node already is initialized with a vector of "" equal to the root length for its annotations, it will just not copy anything and leave it at the default ["",""] if the current node is somehow missing annotations, or is short an annotation.

I'll note that the implementation may break again if somehow a descendant node has MORE annotations than the root. This is because nodes are initialized with a fixed number of empty annotations, instead of an empty vector, and we never add to or subtract from the vector (really, the vector is not even really the right data structure, just an array would serve for having a static number of annotations assigned to every node). Trying to set the third element of the an annotation vector when the root only has two and two were assigned to the new node is problematic. However, this doesn't seem to be an issue that's come up, and the simple fix resolves the case where the node has less annotations than the root.

I've added this to my fork. I've attached the output JSON that results from my fixed implementation. It'll be included on my next PR.

[output.json.gz](https://github.com/yatisht/usher/files/8735356/output.json.gz)
